pulp_smash/tests/docker/api_v2/test_tags.py

```diff
@@ -0,0 +1,228 @@
+# coding=utf-8
+"""Tests that work with tags on docker repositories."""
+import hashlib
+import random
+import unittest
+import uuid
+from urllib.parse import urljoin
+
+from packaging.version import Version
+
+from pulp_smash import api, config, utils
+from pulp_smash.constants import (
+    CONTENT_UPLOAD_PATH,
+    DOCKER_UPSTREAM_NAME,
+    DOCKER_V1_FEED_URL,
+    DOCKER_V2_FEED_URL,
+    REPOSITORY_PATH,
+)
+from pulp_smash.exceptions import TaskReportError
+from pulp_smash.tests.docker.api_v2.utils import gen_repo
+from pulp_smash.tests.docker.utils import set_up_module
+
+
+def setUpModule():  # pylint:disable=invalid-name
+    """Skip tests on Pulp versions lower than 2.12."""
+    set_up_module()
+    if config.get_config().version < Version('2.12'):
+        raise unittest.SkipTest('These tests require at least Pulp 2.12.')
+
+
+def create_docker_repo(cfg, upstream_name, use_v1=False):
+    """Create a docker repository.
+
+    :param pulp_smash.config.ServerConfig cfg: Information about a Pulp host.
+    :param upstream_name: The Docker container upstream name.
+    :param use_v1: If ``True`` use Docker V1 feed URL else use Docker V2 feed
+        URL.
+    :return: Detailed information about the created repository.
+    """
+    body = gen_repo()
+    if use_v1:
+        body['importer_config'].update({
+            'enable_v1': True,
+            'feed': DOCKER_V1_FEED_URL,
+        })
+    else:
+        body['importer_config'].update({
+            'feed': DOCKER_V2_FEED_URL,
+        })
+    body['importer_config']['upstream_name'] = upstream_name
+    client = api.Client(cfg, api.json_handler)
+    return client.post(REPOSITORY_PATH, body)
+
+
+def search_units(cfg, repo, criteria):
```

Comment:
I really need to move `pulp_smash.tests.rpm.api_v2.utils.find_units` to `pulp_smash.utils`.